Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -338,6 +340,17 @@ void ComplexTextController::collectComplexTextRuns()
     // would need to be updated accordingly too.
     nextFont = m_font.fontForCombiningCharacterSequence(baseOfString.first(currentIndex));
 
+    auto getHalfWidthFontIfNeeded = [&] (RefPtr<const Font>& protectedHalfWidthFont, TextSpacing::CharactersData& charactersData) {
```

Comment:
Do we really need `protectedHalfWidthFont` and/or have it passed by reference here? Shouldn't whatever pointer we return here be responsible by the caller for protecting it (which we seem to do here already AFAICT)?